tests/dataimportcron_test.go

```diff
@@ -0,0 +1,225 @@
+package tests_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	cdiclientset "kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned"
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	dataImportCronTimeout = 90 * time.Second
+)
+
+var (
+	importsToKeep int32 = 1
+)
+
+var _ = Describe("DataImportCron", func() {
+	const (
+		scheduleEveryMinute = "* * * * *"
+		scheduleOnceAYear   = "0 0 1 1 *"
+	)
+	var (
+		f                   = framework.NewFramework(namespacePrefix)
+		registryPullNode    = cdiv1.RegistryPullNode
+		trustedRegistryURL  = func() string { return fmt.Sprintf(utils.TrustedRegistryURL, f.DockerPrefix) }
+		externalRegistryURL = "docker://quay.io/kubevirt/cirros-container-disk-demo"
+		cron                *cdiv1.DataImportCron
+		err                 error
+	)
+
+	AfterEach(func() {
+		if cron != nil {
+			By("Delete cron")
+			err = DeleteDataImportCron(f.CdiClient, cron.Namespace, cron.Name)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	table.DescribeTable("should", func(schedule string, repeat int, checkGarbageCollection bool) {
+		var url string
+
+		if repeat > 1 || utils.IsOpenshift(f.K8sClient) {
```

Comment:
Shouldn't ever go to the 'net for functional tests.  Will this be addressed later?

Reply:
Yes we will. Adding it to the post-merge tasks.

Reply:
Well the problem here is we don't have a good repo with these D/S. We need to brain storm a little bit on how we can make it work reliable both U/S and D/S without hitting the outside world.